core/primitives-core/src/account.rs

```diff
@@ -114,6 +151,15 @@ impl Account {
         Self::V1(AccountV1 { amount, locked, code_hash, storage_usage })
     }
 
+    pub fn new_v2(
```

Comment:
we don't need to have 2 separate constructors, we can decide based on `AccountContract`, the same way we do in `set_contract`